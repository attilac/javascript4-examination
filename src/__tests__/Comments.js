import React from 'react';
import { mount } from 'enzyme';
import Comments from '../components/Comments';

function postMockComment(){
  const mockComment = JSON.stringify(
    [
      {
        comment: 'testComment',
        id: '1',
        postId: '1',
        author: 'testAuthor',
        date: '2018-1-23 14:44:38'
      }
    ]
  );
  localStorage.setItem('comments', mockComment);
}

beforeEach(() => {
  postMockComment();
});

afterEach(() => {
  localStorage.clear();
});

it('should fetch comment as state on mount', ()=> {
  const wrapper = mount(<Comments postId="1" currentPersona="testAuthor" />)
  expect(wrapper.state().comments[0].comment).toEqual('testComment')
})

it('should remove comment from state', ()=> {
  const wrapper = mount(<Comments postId="1" currentPersona="testAuthor" />)
  expect(wrapper.state().comments).toHaveLength(1);
  wrapper.instance().removeComment("1");
  expect(wrapper.state().comments).toHaveLength(0);
})

it('should render my comment & author', ()=> {
  const wrapper = mount(<Comments postId="1" currentPersona="testAuthor" />)
  expect(wrapper.find('SingleComment div p .text-grey-darker').text()).toEqual('testComment');
  expect(wrapper.find('SingleComment div p .text-grey-dark').text()).toContain('testAuthor');
})

it('should match snapshot', () => {
  const wrapper = mount(<Comments postId="1" currentPersona="testAuthor" />)
  expect(wrapper).toMatchSnapshot();
})


