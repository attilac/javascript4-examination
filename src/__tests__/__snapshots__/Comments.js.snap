// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Comments
  currentPersona="testAuthor"
  postId="1"
>
  <div
    className="py-2"
  >
    <h2
      className="text-indigo-darker border-b mb-4"
    >
      Comments
    </h2>
    <SingleComment
      author="testAuthor"
      comment="Comment"
      currentPersona="testAuthor"
      date="2018-1-23 14:44:38"
      id="1"
      key="1"
      onClick={[Function]}
      postId="1"
    >
      <div
        className="py-2 border-b relative"
        key="1"
      >
        <Button
          className="absolute pin-t pin-r my-3"
          danger={true}
          onClick={[Function]}
        >
          <button
            className="bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r my-3"
            data-test="button"
            onClick={[Function]}
          >
            x
          </button>
        </Button>
        <p
          className="text-grey-darker mb-4"
        >
          Comment
        </p>
        <p
          className="text-grey-dark mb-4"
        >
          Posted by: 
          testAuthor
           @ 
          2018-1-23 14:44:38
        </p>
      </div>
    </SingleComment>
    <CreateNewComment
      author="testAuthor"
      postId="1"
      updateComments={[Function]}
    >
      <form
        className="container mx-auto flex flex-col p-6"
        onSubmit={[Function]}
      >
        <label
          className="mb-2"
          htmlFor="comment"
        >
          Comment
        </label>
        <textarea
          className="shadow rounded border mb-2 p-2"
          id="comment"
          name="comment"
          onChange={[Function]}
          value=""
        />
        <input
          className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
          type="submit"
          value="Comment"
        />
      </form>
    </CreateNewComment>
  </div>
</Comments>
`;
